EventFiltering/PWGCF/CFFilterTwoN.cxx

```diff
@@ -0,0 +1,345 @@
+    auto partsAntiL = partAntiL->sliceByCached(aod::femtodreamparticle::femtoDreamCollisionId, col.globalIndex());
+
+    auto magneticField = col.magField();
+    registry.fill(HIST("fProcessedEvents"), 0);
+    registry.fill(HIST("fMultiplicityBefore"), col.multV0M());
+    registry.fill(HIST("fZvtxBefore"), col.posZ());
+
+    int Nproton = 0;
+    int Nantiproton = 0;
+    int Nlambda = 0;
+    int Nantilambda = 0;
+    int Ndeuteron = 0;
+    int Nantideuteron = 0;
+
+    for (auto pd : partsPD) {
+      registry.fill(HIST("fPtBeforeSel"), pd.pt());
+      // select protons
+      if (KstarTrigger.value == 0 || KstarTrigger.value == 11) {
+        if (SelectParticlePID(pd.pidcut(), o2::track::PID::Proton, pd.p(), confPIDThreshold)) {
+          registry.fill(HIST("fPtProtonAfterSel"), pd.pt());
+          Nproton++;
+        }
+      }
+      // select deuterons
+      if (SelectParticlePID(pd.pidcut(), o2::track::PID::Deuteron, pd.p(), confPIDThreshold)) {
+        registry.fill(HIST("fPtDeuteronAfterSel"), pd.pt());
+        Ndeuteron++;
+      }
+    }
+    if (KstarTrigger.value == 1 || KstarTrigger.value == 11) {
+      for (auto lambda : partsL) {
+        registry.fill(HIST("fPtLambdaAfterSel"), lambda.pt());
+        Nlambda++;
+      }
+    }
+
+    for (auto antipd : partsAntiPD) {
+      registry.fill(HIST("fPtAntiBeforeSel"), antipd.pt());
+      // select antiprotons
+      if (KstarTrigger.value == 0 || KstarTrigger.value == 11) {
+        if (SelectParticlePID(antipd.pidcut(), o2::track::PID::Proton, antipd.p(), confPIDThreshold)) {
+          registry.fill(HIST("fPtAntiProtonAfterSel"), antipd.pt());
+          Nantiproton++;
+        }
+      }
+      // select antideuterons
+      if (SelectParticlePID(antipd.pidcut(), o2::track::PID::Deuteron, antipd.p(), confPIDThreshold)) {
+        registry.fill(HIST("fPtAntiDeuteronAfterSel"), antipd.pt());
+        Nantideuteron++;
+      }
+    }
+
+    if (KstarTrigger.value == 1 || KstarTrigger.value == 11) {
+      for (auto antilambda : partsAntiL) {
+        registry.fill(HIST("fPtAntiLambdaAfterSel"), antilambda.pt());
+        Nlambda++;
+      }
+    }
+
+    bool keepEvent[kLAST_CFTwoBodyTriggers] = {false, false};
+    int lowKstarPairs[kLAST_CFTwoBodyTriggers] = {0, 0};
+
+    // trigger for pd pairs
+    if (KstarTrigger.value == 0 || KstarTrigger.value == 11) {
+      if (Ndeuteron > 0 && Nproton > 0) {
+        for (auto& [p1, p2] : combinations(soa::CombinationsStrictlyUpperIndexPolicy(partsPD, partsPD))) {
+          if (
+            !(SelectParticlePID(p1.pidcut(), o2::track::PID::Proton, p1.p(), confPIDThreshold.value) &&
+              SelectParticlePID(p2.pidcut(), o2::track::PID::Deuteron, p2.p(), confPIDThreshold.value)) &&
+            !(SelectParticlePID(p1.pidcut(), o2::track::PID::Deuteron, p1.p(), confPIDThreshold.value) &&
+              SelectParticlePID(p2.pidcut(), o2::track::PID::Proton, p2.p(), confPIDThreshold.value))) {
+            continue;
```

Comment:
this checks for wither p-d or d-p pair but the later code calculates k star assuming p-d pair. You should calculate kstar accordingly too which of these if statements is correct

Reply:
fixed.